protected/tests/unit/CurationLogTest.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+class CurationLogTest extends CDbTestCase
+{
+    protected $fixtures=array(
+        'log'=>'CurationLog',
+        'dataset'=>'Dataset',
+    );
+
+    public function testCreateLogEntry()
+    {
+        $this->assertEquals(date("Y-m-d"), $this->log(0)->creation_date, "Different date format with CurationLog");
```

Comment:
I think you need to call the static `createLogEntry($id)` function in the `CurationLog` model class from here. If the `createLogEntry($id)` function returns `false` because of some problem with saving the new curation log entry then you can test for a boolean value here.

Reply:
I am trying to make this code working from the `/models/CurationLog.php/`
```
    public static function createCurationLogEntry($id)
    {
        $model = Dataset::model()->findByPk($id);
        if ($model->upload_status === "Published") {
            $curationlog = new CurationLog;
            $curationlog->creation_date = date("Y-m-d");
            $curationlog->last_modified_date = null;
            $curationlog->dataset_id = $id;
            $curationlog->created_by = "System";
            $curationlog->action = "Status changed to stuff";
            if (!$curationlog->save())
                return false;
        }
``` 
When I click `Delete` in the `file attribute` page, there is no entry in the tbl `curation_log`, am I missing something in the configuration or calling that function not in proper way?

Reply:
Have you called this `createCurationLogEntry($id)` function from somewhere after the file attribute is deleted?

Reply:
It looks like you are deleting the file attribute in `DeleteFileAttributeAction` class. Try calling the `createCurationLogEntry($id)` function after `$attribute->delete();` to create a new entry in curation_log table after the file attribute is deleted.

Reply:
> ```
> CurationLog::createCurationLogEntry("8");
> ```
This is working. But the `id` is hardcoded, I will try to replace it.

Reply:
The hardcoded `id` has been replaced.